lxd/container_lxc.go

```diff
@@ -5695,13 +5700,29 @@ func (c *containerLXC) createUnixDevice(m types.Device) ([]string, error) {
 		}
 	}
 
+	relativeSrcPath := strings.TrimPrefix(m["source"], "/")
+
+	// legacy style unix device creation
+	devName := fmt.Sprintf("unix.%s", strings.Replace(relativeSrcPath, "/", "-", -1))
+	devPathLegacy := filepath.Join(c.DevicesPath(), devName)
+
+	destPath, exist := m["path"]
+	if !exist {
+		destPath = m["source"]
+	}
+	relativeDestPath := strings.TrimPrefix(destPath, "/")
+
+	// new style unix device creation
+	devName = fmt.Sprintf("unix.%s.%s", name, strings.Replace(relativeDestPath, "/", "-", -1))
+	devPathDeviceNameEncoded := filepath.Join(c.DevicesPath(), devName)
+
```

Comment:
So that's changing the cleanup code to clear the old device files but what cleans up the new ones now?

Reply:
Ah right, this is something I wanted to ask. Why is this old codepath needed. It seems weird to me to have it in the first place.

Reply:
Something needs to cleanup the mess if the container isn't shut down cleanly. But maybe we have some other code that takes care of that these days.